pkg/terraformdocsgenerator/openapiterraformdocsgenerator/openapi_terraform_provider_doc_generator.go

```diff
@@ -1,33 +1,61 @@
+	"log"
 	"sort"
 )
 
 // TerraformProviderDocGenerator defines the struct that holds the configuration needed to be able to generate the documentation
 type TerraformProviderDocGenerator struct {
 	// ProviderName defines the provider name
 	ProviderName string
+	// Hostname the Terraform registry that distributes the provider as documented in https://www.terraform.io/docs/language/providers/requirements.html#source-addresses
+	// For in-house providers that you intend to distribute from a local filesystem directory, you can use an arbitrary hostname in a domain your organization controls. For example, if your corporate domain were example.com then you might choose
+	// to use terraform.example.com as your placeholder hostname, even if that hostname doesn't actually resolve in DNS.
+	Hostname string
+	// Namespace An organizational namespace within the specified registry to be used for configuration purposes as documented in https://www.terraform.io/docs/language/providers/requirements.html#source-addresses
+	Namespace string
+	// PluginVersionConstraint should contain the OpenAPI plugin version constraint eg: "~> 2.1.0". If not populated the renderer
+	// will default to ">= 2.1.0" OpenAPI provider version
+	PluginVersionConstraint string
 	// SpecAnalyser analyses the swagger doc and provides helper methods to retrieve all the end points that can
 	// be used as terraform resources.
 	SpecAnalyser openapi.SpecAnalyser
 }
 
 // NewTerraformProviderDocGenerator returns a TerraformProviderDocGenerator populated with the provider documentation which
 // exposes methods to render the documentation in different formats (only html supported at the moment)
-func NewTerraformProviderDocGenerator(providerName, openAPIDocURL string) (TerraformProviderDocGenerator, error) {
+func NewTerraformProviderDocGenerator(providerName, hostname, namespace, openAPIDocURL string) (TerraformProviderDocGenerator, error) {
 	analyser, err := openapi.CreateSpecAnalyser("v2", openAPIDocURL)
 	if err != nil {
 		return TerraformProviderDocGenerator{}, err
 	}
-	return TerraformProviderDocGenerator{ProviderName: providerName, SpecAnalyser: analyser}, nil
+	return TerraformProviderDocGenerator{
+		ProviderName: providerName,
+		Hostname:     hostname,
+		Namespace:    namespace,
+		SpecAnalyser: analyser,
+	}, nil
 }
 
 // GenerateDocumentation creates a TerraformProviderDocumentation object populated based on the OpenAPIDocURL documentation
 func (t TerraformProviderDocGenerator) GenerateDocumentation() (TerraformProviderDocumentation, error) {
+	if t.ProviderName == "" {
+		return TerraformProviderDocumentation{}, errors.New("provider name not provided")
+	}
+	if t.Hostname == "" {
+		return TerraformProviderDocumentation{}, errors.New("hostname name not provided, this is required to be able to render the provider installation section containing the required_providers block with the source address configuration in the form of [<HOSTNAME>/]<NAMESPACE>/<TYPE>")
```

Comment:
Is the `name` here after `hostname` a typo? 

```suggestion
		return TerraformProviderDocumentation{}, errors.New("hostname not provided, this is required to be able to render the provider installation section containing the required_providers block with the source address configuration in the form of [<HOSTNAME>/]<NAMESPACE>/<TYPE>")
```

Reply:
Yep, pushing a commit with a fix shortly